### PR TITLE
ARROW-7479: [Rust][Ruby][R] Fix typos

### PR DESCRIPTION
**Pull request by kiszk**
This PR fixes typos in files under `rust`, `ruby`, and `r` directories

### CODE REVIEW

File: r/man/read_delim_arrow.Rd [hunk 1]
```diff
@@ -109,7 +109,7 @@ parsing options provided in other arguments (e.g. \code{delim}, \code{quote}, et
 \link[=Table]{arrow::Table}?}
 }
 \value{
-A \code{data.frame}, or an Table if \code{as_data_frame = FALSE}.
+A \code{data.frame}, or a Table if \code{as_data_frame = FALSE}.
```
Comment (nealrichardson) [inline r/man/read_delim_arrow.Rd]: This is a generated file, so you should fix the source too: https://github.com/apache/arrow/blob/master/r/R/csv.R#L70

**Review by paddyhoran — CHANGES_REQUESTED** [review]
2 that look like they should be `and`, all others look good from a Rust perspective.  Thanks @kiszk

File: rust/arrow/src/ipc/gen/SparseTensor.rs [hunk 2]
```diff
@@ -89,7 +89,7 @@ pub enum SparseTensorIndexCOOOffset {}
 
 /// ----------------------------------------------------------------------
 /// EXPERIMENTAL: Data structures for sparse tensors
-/// Coodinate (COO) format of sparse tensor index.
+/// Coordinate (COO) format of sparse tensor index.
```
Comment (paddyhoran) [inline rust/arrow/src/ipc/gen/SparseTensor.rs]: Also, I think this is generated

File: r/man/read_delim_arrow.Rd (hunk 1 above)
Comment (kiszk) [reply r/man/read_delim_arrow.Rd]: Thank you for your comment. I addressed this.

File: rust/arrow/src/ipc/file/reader.rs [hunk 3]
```diff
@@ -305,7 +305,7 @@ fn create_primitive_array(
     make_array(array_data)
 }
 
-/// Reads the correct number of buffers based on list type an null_count, and creates a
+/// Reads the correct number of buffers based on list type a null_count, and creates a
```
[the remark this answers is not in the document]
Comment (kiszk) [reply rust/arrow/src/ipc/file/reader.rs]: Thank you for your comments. I addressed them.

File: rust/arrow/src/ipc/gen/SparseTensor.rs (hunk 2 above)
Comment (kiszk) [reply rust/arrow/src/ipc/gen/SparseTensor.rs]: I already updated `format/SparseTensor.fbs`. Is there any another file to generate this?

Comment (paddyhoran) [reply rust/arrow/src/ipc/gen/SparseTensor.rs]: No, that should be it I think.  Thanks

**Review by paddyhoran — APPROVED** [review]
LGTM for the Rust changes

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**